batch/batch/front_end/front_end.py

```diff
@@ -284,9 +285,13 @@ async def _query_batch_jobs(request, batch_id):
   FROM jobs
   INNER JOIN batches ON jobs.batch_id = batches.id
   LEFT JOIN job_attributes
-  ON jobs.batch_id = job_attributes.batch_id AND
-    jobs.job_id = job_attributes.job_id AND
-    job_attributes.`key` = 'name'
+    ON jobs.batch_id = job_attributes.batch_id AND
+      jobs.job_id = job_attributes.job_id AND
+      job_attributes.`key` = 'name'
+  LEFT JOIN batch_updates
+    ON jobs.batch_id = batch_updates.batch_id AND
+      jobs.job_id >= batch_updates.start_job_id AND
+      jobs.job_id < batch_updates.start_job_id + batch_updates.n_jobs
```

Comment:
This will scan every batch_update whose start id is before this job id, for the given batch. Ergo, this page is now O(n_updates), not O(1).

You haven't told the database that these are disjoint intervals so it can't do anything smarter. I think the only way to make this efficient is to store the update id in the job record. What the arguments against putting the update id in the job table?

Reply:
I don't believe we've had this discussion, but I think that's a very reasonable approach and more intuitive than looking at the range.

Reply:
Just make sure if you need to add an index on the jobs table for the update id that it will be instant. I’m guessing it’s probably not and will shut batch down for at least an hour.

Reply:
aren't y'all supposed to be on vacation